druid/examples/custom_widget.rs

```diff
@@ -81,21 +84,18 @@ impl Widget<String> for CustomWidget {
         let fill_color = Color::rgba8(0x00, 0x00, 0x00, 0x7F);
         ctx.fill(rect, &fill_color);
 
-        // Text is easy, if you ignore all these unwraps. Just pick a font and a size.
-        // Here's where we actually use the UI state
-        let layout = ctx
-            .text()
-            .new_text_layout(data.as_str())
-            .font(FontFamily::SYSTEM_UI, 24.0)
-            .text_color(fill_color)
-            .build()
-            .unwrap();
+        // Text is easy; in real use TextLayout should be stored in the widget
+        // and reused.
+        let mut layout = TextLayout::new(data.as_str());
+        layout.set_font(FontDescriptor::new(FontFamily::SERIF).with_size(24.0));
+        layout.set_text_color(fill_color);
+        layout.rebuild_if_needed(&mut ctx.text(), env);
 
         // Let's rotate our text slightly. First we save our current (default) context:
         ctx.with_save(|ctx| {
             // Now we can rotate the context (or set a clip path, for instance):
             ctx.transform(Affine::rotate(0.1));
-            ctx.draw_text(&layout, (80.0, 40.0));
+            layout.draw(ctx, (80.0, 40.0));
```

Comment:
Not going to suggest it for this PR, but seeing this makes me wonder whether we might add some ext methods on our `PaintCtx` to make this a little smoother.